src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -386,13 +386,15 @@ private void searchTopEntitiesForMultiCategoryHC(
             logger.error("Failed to get top entities for detector " + adTask.getDetectorId(), e);
             internalHCListener.onFailure(e);
         });
+
+        int minimumDocCount = Math.min((int) (bucketInterval / adTask.getDetector().getDetectorIntervalInMilliseconds()), 1);
```

Comment:
This would choose entities that appear in every detector interval.  Is this what you want?  It sounds too restricted.

Reply:
Good catch, forgot to add * 0.5 , will tune this